arangod/Metrics/ClusterMetricsFeature.cpp

```diff
@@ -0,0 +1,192 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Valery Mironov
+////////////////////////////////////////////////////////////////////////////////
+
+#include "Metrics/ClusterMetricsFeature.h"
+
+#include "ApplicationFeatures/ApplicationServer.h"
+#include "Basics/debugging.h"
+#include "Cluster/ClusterFeature.h"
+#include "Cluster/ClusterMethods.h"
+#include "Cluster/ServerState.h"
+#include "Containers/FlatHashSet.h"
+#include "Logger/LoggerFeature.h"
+#include "Metrics/Metric.h"
+#include "Metrics/MetricsFeature.h"
+#include "Network/NetworkFeature.h"
+#include "ProgramOptions/ProgramOptions.h"
+#include "ProgramOptions/Section.h"
+#include "RestServer/QueryRegistryFeature.h"
+#include "RocksDBEngine/RocksDBEngine.h"
+#include "Scheduler/SchedulerFeature.h"
+#include "Statistics/StatisticsFeature.h"
+#include "StorageEngine/EngineSelectorFeature.h"
+
+namespace arangodb::metrics {
+using namespace std::chrono_literals;
+
+ClusterMetricsFeature::ClusterMetricsFeature(Server& server)
+    : ArangodFeature{server, *this} {
+  setOptional();
+  startsAfter<ClusterFeature>();
+  startsAfter<NetworkFeature>();
+  startsAfter<SchedulerFeature>();
+}
+
+void ClusterMetricsFeature::collectOptions(
+    std::shared_ptr<options::ProgramOptions> options) {
+  options
+      ->addOption(
+          "--server.cluster-metrics-timeout",
+          "Seconds that will be between cross cluster calls that collect "
```

Comment:
maybe something like `Cluster metrics polling timeout in seconds`?

Reply:
Ok